### PR TITLE
use goroutines to accelerate Azure deployments

### PR DESCRIPTION
**Pull request by ypwong99**
This PR parallelises and accelerates the deployment and removal of Azure services with the use of goroutines.

A random five letter prefix is also generated for the Azure service names with `GenerateRandLowercaseLetters` to reduce the chances of deployment conflicts.

### CODE REVIEW

File: src/setup/run.go [hunk 1]
```diff
@@ -145,6 +146,11 @@ func ProvisionFunctionsServerlessAWS(config *Configuration, serverlessDirPath st
 }
 
 func ProvisionFunctionsServerlessAzure(config *Configuration, serverlessDirPath string) {
+	mu := sync.Mutex{}
+	wg := sync.WaitGroup{}
+
+	randomExperimentTag := util.GenerateRandLowercaseLetters(5)
```
Comment (kk-min) [inline src/setup/run.go:152]: What is the purpose of the randomized tag? Is parallelism and subexperiment number not sufficient to differentiate?

**Review by kk-min — APPROVED** [review]
LGTM, left a clarification question

File: src/setup/run.go (hunk 1 above)
Comment (ypwong99) [reply src/setup/run.go:152]: Previously I faced an issue where two experiments running concurrently (e.g. two end to end tests for a PR) were both trying to interact with the same resource group (like `sls-wus-dev-stellar-azure-subex0-para0-rg`)

Comment (kk-min) [reply src/setup/run.go:152]: Makes sense!

**Comment by kk-min** [comment]
@ypwong99 Could you help to fast-forward the `continuous-benchmarking` branch to sync with the changes in this PR as well? Thanks!

**Comment by ypwong99** [comment]
> @ypwong99 Could you help to fast-forward the `continuous-benchmarking` branch to sync with the changes in this PR as well? Thanks!

Sure, btw `continuous-benchmarking-new` and `continuous-benchmarking-test` are no longer in use right?

**Comment by kk-min** [comment]
> > @ypwong99 Could you help to fast-forward the `continuous-benchmarking` branch to sync with the changes in this PR as well? Thanks!
> 
> Sure, btw `continuous-benchmarking-new` and `continuous-benchmarking-test` are no longer in use right?

Yup they're no longer required.

**Pull request merged**